packages/@aws-cdk/cdk/lib/resource-identifiers.ts

```diff
@@ -0,0 +1,50 @@
+import { ArnComponents } from './arn';
+import { CrossEnvironmentPhysicalArnToken, CrossEnvironmentPhysicalNameToken } from './cross-environment-token';
+import { Resource } from './resource';
+
+/**
+ * Construction properties for {@link ResourceIdentifiers}.
+ */
+export interface ResourceIdentifiersProps {
+  /**
+   * The {@link Resource} instance we're defining identifiers for.
+   */
+  readonly resource: Resource;
```

Comment:
feels like this could be the first positional argument instead of a prop

Reply:
Changed.